src/t_hash.c

```diff
@@ -778,12 +1070,25 @@ robj *hashTypeDup(robj *o, sds newkey, uint64_t *minHashExpire) {
     serverAssert(o->type == OBJ_HASH);
 
     if(o->encoding == OBJ_ENCODING_LISTPACK) {
-        unsigned char *zl = o->ptr;
+        unsigned char *zl = hashLpGetListpack(o->ptr);
         size_t sz = lpBytes(zl);
         unsigned char *new_zl = zmalloc(sz);
         memcpy(new_zl, zl, sz);
-        hobj = createObject(OBJ_HASH, new_zl);
-        hobj->encoding = OBJ_ENCODING_LISTPACK;
+
+        if (!hashLpIsHfe(o->ptr)) {
+            hobj = createObject(OBJ_HASH, new_zl);
+            hobj->encoding = OBJ_ENCODING_LISTPACK;
+        } else {
+            listpackHfe *hfe = hashLpGet(o->ptr);
+            if (hfe->meta.trash == 0)
+                *minHashExpire = ebGetMetaExpTime(&hfe->meta);
+
+            listpackHfe *hfe2 = zcalloc(sizeof(*hfe2));
+            hfe2->lp = new_zl;
```

Comment:
```suggestion
            listpackHfe *hfe2 = zcalloc(sizeof(*hfe2));
            hfe2->lp = new_zl;
            hfe2->key = newkey;
```

My next commit will take care that `copyCommand` will pass `newkey` instance, the same one that stored is stored in keyspace.